sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/SubscriptionPartitionLoader.java

```diff
@@ -0,0 +1,102 @@
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.apache.beam.sdk.transforms.Watch;
+import org.apache.beam.sdk.transforms.Watch.Growth.PollFn;
+import org.apache.beam.sdk.transforms.Watch.Growth.PollResult;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PBegin;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TypeDescriptor;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableList;
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+
+class SubscriptionPartitionLoader extends PTransform<PBegin, PCollection<SubscriptionPartition>> {
+  private final TopicPath topic;
+  private final SubscriptionPath subscription;
+  private final SerializableFunction<TopicPath, Integer> getPartitionCount;
+  private final Duration pollDuration;
+  private final boolean terminate;
+
+  SubscriptionPartitionLoader(TopicPath topic, SubscriptionPath subscription) {
+    this(
+        topic,
+        subscription,
+        PartitionLookupUtils::numPartitions,
+        Duration.standardMinutes(1),
+        false);
+  }
+
+  @VisibleForTesting
+  SubscriptionPartitionLoader(
+      TopicPath topic,
+      SubscriptionPath subscription,
+      SerializableFunction<TopicPath, Integer> getPartitionCount,
+      Duration pollDuration,
+      boolean terminate) {
+    this.topic = topic;
+    this.subscription = subscription;
+    this.getPartitionCount = getPartitionCount;
+    this.pollDuration = pollDuration;
+    this.terminate = terminate;
+  }
+
+  @Override
+  public PCollection<SubscriptionPartition> expand(PBegin input) {
+    PCollection<TopicPath> start = input.apply(Create.of(ImmutableList.of(topic)));
+    PCollection<KV<TopicPath, Partition>> partitions =
+        start.apply(
+            Watch.growthOf(
+                    new PollFn<TopicPath, Partition>() {
+                      @Override
+                      public PollResult<Partition> apply(TopicPath element, Context c) {
+                        checkArgument(element.equals(topic));
+                        int partitionCount = getPartitionCount.apply(element);
+                        List<Partition> partitions =
+                            IntStream.range(0, partitionCount)
+                                .mapToObj(Partition::of)
+                                .collect(Collectors.toList());
+                        return PollResult.incomplete(Instant.EPOCH, partitions);
+                      }
+                    })
+                .withPollInterval(pollDuration)
+                .withTerminationPerInput(
+                    terminate
+                        ? Watch.Growth.afterTotalOf(pollDuration.multipliedBy(10))
```

Comment:
Even though the `terminate` will not be `true` in this PR, I'm wondering how is `pollDuration.multipliedBy(10)` computed?

Reply:
This is for unit testing purposes only. If set to true, it polls 10 times.